modules/nw-bgp-examples.adoc

```diff
@@ -0,0 +1,259 @@
+== Advertising pod IPs from a user-defined network over BGP
+
+In this scenario, the blue user-defined network is exposed to the external network so that the attached pods have IP addresses that are reachable from the external network. The user-defined networks are each attached to a different VRF device:
+
+Red user-defined network::
+- A VRF named `mp0-red`
+- A subnet of `10.0.0.0/24`
+
+Blue user-defined network::
+- A VRF named `mp0-blue`
+- A subnet of `10.0.1.0/24`
+
+In this configuration, two separate user-defined networks are defined. The red network covers the `10.0.0.0/24` subnet and the blue network covers the `10.0.1.0/24` subnet.
+
+The following `RouteAdvertisements` CR describes the configuration for the blue tenant:
+
+.`RouteAdvertisements` CR for the blue tenant
+[source,yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: RouteAdvertisements
+metadata:
+  name: default
+spec:
+  networkSelector:
+    matchLabels:
+      k8s.ovn.org/metadata.name: blue
+  frrConfigurationSelector:
+    matchLabels:
+      routeAdvertisements: default
+----
+
+When the OVN-Kubernetes controller sees this `RouteAdvertisements` CR, it generates a `FRRConfiguration` object that configures the FRR daemon to advertise the routes.
+
+.`FRRConfiguration` CR generated by OVN-Kubernetes
+[source,yaml]
+----
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: route-advertisements-blue
+  namespace: openshift-frr-k8s
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      vrf: blue
+      prefixes:
+      - 10.0.1.0/24
+    - asn: 64512
+      neighbors:
+        - address: 172.18.0.5
+          asn: 64512
+          toAdvertise:
+            allowed:
+              prefixes:
+              - 10.0.1.0/24
+  raw:
+    rawConfig: |-
+      router bgp 64512
+      address-family ipv4 unicast
+        import vrf blue
+        exit-address-family
+      router bgp 64512 vrf blue
+      address-family ipv4 unicast
+        import vrf default
+        exit-address-family
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/hostname: ovn-worker
+----
+
+The generated `FRRConfiguration` object configures the subnet `10.0.1.0/24`, which belongs to VRF blue, is imported into the default VRF and advertised to the `172.18.0.5` neighbor. Because the `targetVRF` uses the default value, the routes are leaked and advertised in the default VRF. Additionally, routes are imported from the default VRF into the blue VRF.
+
+[id="advertising-pod-ips-from-a-user-defined-network-over-bgp-with-vpn_{context}"]
+== Advertising pod IPs from a user-defined network over BGP with VPN
+
+In this scenario, a VLAN interface attached to a VRF device is associated with the blue network and the VLAN includes the external PE router. This setup provides a _VRF lite_ design where FRR-K8S is leveraged to advertise the blue network over only the corresponding VRF/VLAN link to the next hop PE router. The red tenant uses the same configuration.
```

Comment:
I would probably rephrase
```
In this scenario, a VLAN interface is attached to the VRF device associated with the blue network. This setup provides ...
```
Additionally we might need to provide some examples on how to configura that VLAN with NMState. I will get back to you on this.